### PR TITLE
Fix usage of inherited Sinatra::Base classes keyword arguments

### LINKED ISSUE #1669: Incompatibility with Ruby 3 when extending Sinatra::Base
Hey Folks 👋 

I'm trying to upgrade an app to test with Ruby 3 and I'm having a problem.

This is the minimal example to reproduce it: 
```
class Foo < Sinatra::Base
  def initialize(app:, comp:)
    p app
    p comp
  end
end
```

The Class Foo extends from Sinatra::Base and I redefine the initializer to accept two named parameter and it fails with this error:
```
 Foo.new(app: 1, comp: 2)
ArgumentError: wrong number of arguments (given 1, expected 0; required keywords: app, comp)
from (pry):3:in `initialize
```

I think the problem is inside the `new` redefinition here https://github.com/sinatra/sinatra/blob/master/lib/sinatra/base.rb#L1525-L1528

```
      def new(*args, &bk)
        instance = new!(*args, &bk)
        Wrapper.new(build(instance).to_app, instance)
      end
```

I tried to fork the sinatra code and change this method to be accepted as the new ruby 3 pattern https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/ 
but I failed without breaking the original behave. 

any ideas on what to do here?

I'm thinking on redefine `new` instead of creating an initializer as a workaround for this.

Thanks

### PR DESCRIPTION
**Pull request by duduribeiro**
Closes #1669 
Using keyword arguments on inherited classes from Sinatra::Base breaks
with ArgumentError: wrong number of arguments (given X, expected 0) on Ruby 3

This can be reproduced by this minimal example:

# app.rb
```
require 'sinatra'

class KWApp < Sinatra::Base
  def initialize(app:, comp:)
    p app
    p comp

    app.get '/' do
      "Hello from KWApp"
    end

    super(app)
  end
end

class PApp < Sinatra::Base
  def initialize(app, comp)
    p app
    p comp

    app.get '/' do
      "Hello from PApp"
    end

    super(app)
  end
end
```

# config.ru
```
require "./app"
require "sinatra"

map "/" do
    run KWApp.new(app: Sinatra.new, comp: 2)
end

map "/p" do
    run PApp.new(Sinatra.new, 2)
end
```

This example works on ruby < 3 but it breaks when using Ruby 3 with:

```
 KWApp.new(app: 1, comp: 2)
ArgumentError: wrong number of arguments (given 1, expected 0; required keywords: app, comp)
from (pry):3:in `initialize
```

This commit fixes the usage of it on Ruby3 by allowing the method to
accept both positional and keyword arguments.

### CODE REVIEW

**Comment by dentarg** [comment]
@duduribeiro Nice! Do you think you could add your example as a test?

**Comment by duduribeiro** [comment]
@dentarg yeah!. Will do it and ping you after.

Thanks 🍻

**Comment by duduribeiro** [comment]
hey @dentarg,
I've added a test of it. Can u check if it is ok?

Thanks

File: .travis.yml
Comment (duduribeiro) [inline .travis.yml:4]: On trusty rvm is failing to install ruby 3.0.0-preview1, but upgrading to xenial makes the jruby build fails.

Comment (dentarg) [reply .travis.yml:4]: I think we should try to get the test suite running on GitHub Actions

Comment (duduribeiro) [reply .travis.yml:4]: @dentarg wanna merge this and work on GH actions after? I can work on this in a new PR if I get access on it.

Also, this will be importante because:
https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
```
Building on a public repositories only
We love our OSS teams who choose to build and test using TravisCI and we fully want to support that community. However, in recent months we have encountered significant abuse of the intention of this offering (increased activity of cryptocurrency miners, TOR nodes operators etc.). Abusers have been tying up our build queues and causing performance reductions for everyone. In order to bring the rules back to fair playing grounds, we are implementing some changes for our public build repositories.
For those of you who have been building on public repositories (on travis-ci.com, with no paid subscription), we will upgrade you to our trial (free) plan with a 10K credit allotment (which allows around 1000 minutes in a Linux environment).
```

Comment (dentarg) [reply .travis.yml:4]: @duduribeiro I'm not maintainer or collaborator, just a fellow contributor :-) It is @namusyaka or @jkowens that can merge PRs

Comment (dentarg) [reply .travis.yml:4]: I think Actions should work in your fork of Sinatra, so you can work on it right now if you want

**Comment by jkowens** [comment]
I've worked out the Travis issues. Should be good if you rebase master. Thanks!

File: .travis.yml
Comment (jkowens) [reply .travis.yml:4]: Switching to Github Actions would be 💯

**Comment by duduribeiro** [comment]
@jkowens thanks!

This PR is ready for review now 🍻

File: lib/sinatra/base.rb
Comment (jkowens) [inline lib/sinatra/base.rb:1526]: Out of curiosity, what happens if we just did this:

```
instance = new!(*args, **kwargs, &bk)
```

I think the `initialize` method signature would need updated, but would that work? Or would that break something?

Comment (duduribeiro) [reply lib/sinatra/base.rb]: hey @jkowens 
thanks for the suggestion!

It worked perfectly!

already pushed the changes

Comment (jkowens) [reply lib/sinatra/base.rb]: I think updating the `initialize` method signature will resolve some of the build failures: https://github.com/sinatra/sinatra/blob/cb64dd3516e486f79b3ecaad2c473830d61e56b5/lib/sinatra/base.rb#L919

```
def initialize(app = nil, **kwargs)
```

Comment (duduribeiro) [reply lib/sinatra/base.rb]: hey @jkowens, now build is breaking because of 
```
Template engine not found: wlang
```

Comment (jkowens) [reply lib/sinatra/base.rb]: Hmm that one doesn't make sense to me!

Comment (jkowens) [reply lib/sinatra/base.rb]: Ah! It looks like wlang 3.0 isn't compatible at the the moment. Can you go to the Gemfile and lock wlang to '~> 2.3.1'?

Comment (jkowens) [reply lib/sinatra/base.rb]: Actually, I'll do that in a separate commit 👍

Comment (jkowens) [reply lib/sinatra/base.rb]: Please rebase master and I think it should be good to go!

**Comment by jkowens** [comment]
This looks good to me! @namusyaka can you take a look as well?

**Review by namusyaka — APPROVED** [review]
Looks good, thanks.

**Pull request merged**
